superset/db_engine_specs/trino.py

```diff
@@ -69,3 +69,37 @@ def adjust_database_uri(
             selected_schema = parse.quote(selected_schema, safe="")
             database = database.split("/")[0] + "/" + selected_schema
             uri.database = database
+
+    @classmethod
+    def update_impersonation_config(
+        cls, connect_args: Dict[str, Any], uri: str, username: Optional[str],
+    ) -> None:
+        """
+        Update a configuration dictionary
+        that can set the correct properties for impersonating users
+        :param connect_args: config to be updated
+        :param uri: URI string
+        :param impersonate_user: Flag indicating if impersonation is enabled
+        :param username: Effective username
+        :return: None
+        """
+        url = make_url(uri)
+        backend_name = url.get_backend_name()
+
+        # Must be Presto connection, enable impersonation, and set optional param
```

Comment:
We want to do this only if "enable impersonation" is true, no? In that case it would be better to add it to `modify_url_for_impersonation`.

Reply:
this is called only if impersonation is enabled 

_// from core.py_
```
if self.impersonate_user:
            self.db_engine_spec.update_impersonation_config(
                connect_args, str(sqlalchemy_url), effective_username
            )
```

modify_url will work only if authentication is disabled - we will have to modify connection args to pass logged In user_name based on the underlying connector - for presto, its `principal_username` and for trino its `user`.